### ISSUE TITLE
Migration: DataTypeNotSupportedError: Data type "string" is not supported in "mysql" database.

### ISSUE THREAD

**Issue by bensquire** [typeorm/typeorm#644]
When my app runs, the schema sync works as expected; all tables are created, with the expected defaults and relationships.

When I run:

    typeorm migrations:run

Then I get the following error:

````
{ DataTypeNotSupportedError: Data type "string" is not supported in "mysql" database.
    at new DataTypeNotSupportedError (/usr/local/lib/node_modules/typeorm/error/DataTypeNotSupportedError.js:16:28)
    at /usr/local/lib/node_modules/typeorm/metadata-builder/EntityMetadataValidator.js:65:27
    at Array.forEach (native)
    at EntityMetadataValidator.validate (/usr/local/lib/node_modules/typeorm/metadata-builder/EntityMetadataValidator.js:62:36)
    at /usr/local/lib/node_modules/typeorm/metadata-builder/EntityMetadataValidator.js:34:74
    at Array.forEach (native)
    at EntityMetadataValidator.validateMany (/usr/local/lib/node_modules/typeorm/metadata-builder/EntityMetadataValidator.js:34:25)
    at Connection.buildMetadatas (/usr/local/lib/node_modules/typeorm/connection/Connection.js:480:33)
    at Connection.<anonymous> (/usr/local/lib/node_modules/typeorm/connection/Connection.js:133:30)
    at step (/usr/local/lib/node_modules/typeorm/connection/Connection.js:32:23)
  name: 'DataTypeNotSupportedError',
  message: 'Data type "string" is not supported in "mysql" database.' }
````

I've not specified a type of "string" in any entity, and my migration script is v.simple as well:

````
import { Connection, EntityManager, MigrationInterface, QueryRunner } from "typeorm";

export class FirstReleaseChanges1499674151123 implements MigrationInterface {

    public async up(queryRunner: QueryRunner): Promise<any> {
        await queryRunner.insert("insurance", {
            cost: 250,
            value: null,
        });

        await queryRunner.insert("insurance", {
            cost: 500,
            value: 2.00,
        });

        await queryRunner.insert("insurance", {
            cost: 1000,
            value: 5.00,
        });

        await queryRunner.insert("immediacy", {
            collectionLead: 60,
            minutes: 90,
        });

        await queryRunner.insert("immediacy", {
            collectionLead: 60,
            minutes: 120,
        });

        await queryRunner.insert("immediacy", {
            collectionLead: 60,
            minutes: 240,
        });
    }

    public async down(queryRunner: QueryRunner): Promise<any> {
        await queryRunner.truncate("insurance");
        await queryRunner.truncate("immediacy");
    }
}
````

Am I missing something? I'm not sure which entity its struggling with, but here is a good entitty sample:

````
import {
    Column, CreateDateColumn, Entity, JoinColumn,
    ManyToOne, OneToOne, PrimaryColumn, UpdateDateColumn,
} from "typeorm";
import { Address } from "./address";
import { BookingStatus } from "./bookingStatus";
import { Immediacy } from "./immediacy";
import { Insurance } from "./insurance";

@Entity()
export class Quote {
    @PrimaryColumn({ generated: false })
    public id: string;

    @JoinColumn()
    @OneToOne((type) => Insurance, (insurance) => insurance.id)
    public insuranceId;

    @JoinColumn()
    @ManyToOne((type) => Immediacy, (immediacy) => immediacy.id)
    public immediacyId;

    @Column({ type: "datetime" })
    public immediacyDate: Date;

    @CreateDateColumn()
    public dtCreated: Date;

    @UpdateDateColumn()
    public dtUpdated: Date;

    @JoinColumn()
    @ManyToOne((type) => Address, (address) => address.id)
    public addressSource: Address;

    @JoinColumn()
    @ManyToOne((type) => Address, (address) => address.id)
    public addressDestination: Address;

    @JoinColumn()
    @ManyToOne((type) => BookingStatus, (bookingStatus) => bookingStatus.id)
    public bookingStatus: BookingStatus;

    @Column({ type: "text" })
    public bookingObject: string;
}

````

Thanks!

P.S - I'm using `0.1.0-alpha.31`

**Comment by woolyninja** [comment]
I'm having the same issue but with a "postgres" connection?

**Comment by GeeWee** [comment]
I also get a similar error message when upgrading from 0.0.11 to 0.1.0-alpha34

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
